www/js/diary/infinite_scroll_list.js

```diff
@@ -137,6 +141,31 @@ angular.module('emission.main.diary.infscrolllist',['ui-leaflet',
     });
   }
 
+  $scope.$on("scroll.infiniteScrollComplete", function() {
+    if ($scope.infScrollControl.callback != undefined) {
+      $scope.infScrollControl.callback();
+      $scope.infScrollControl.callback = undefined;
+    }
+    $scope.infScrollControl.scrollLock = false;
+  });
+
+  $scope.handleScroll = function() {
+    if ($scope.infScrollControl.scrollLock) return;
+    if ($scope.infScrollControl.reachedEnd) {
+      console.log("reached end");
+      return;
+    }
+    if ($ionicScrollDelegate.getScrollPosition().top < $scope.infScrollControl.threshold) {
+      $scope.infScrollControl.callback = function() {
+        // This whole "infinite scroll upwards" implementation is quite hacky, but after hours of work on it, it's the only way I could approximate the desired behavior.
+        $ionicScrollDelegate.scrollBottom();
+        const clientHeight = $ionicScrollDelegate.getScrollView().__clientHeight;
+        $ionicScrollDelegate.scrollBy(0, -$scope.infScrollControl.fromBottom+clientHeight);
```

Comment:
I can't believe you actually implemented this! I would have given up after the nth hour of fiddly work.